sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1414,6 +1414,22 @@ object functions {
    */
   def md5(columnName: String): Column = md5(Column(columnName))
 
+  /**
+   * Calculates the SHA-2 family of hash functions and returns the value as a hex string.
+   *
+   * @group misc_funcs
+   * @since 1.5.0
+   */
+  def sha2(e: Column, bit: Int): Column = Sha2(e.expr, lit(bit).expr)
```

Comment:
The handling of invalid bit length is useful for SQL (we can not easily valid the parameter in SQL parser), but for DataFrame API, it's good to fail fast, give better feedback to users.